Userland/Libraries/LibWeb/HTML/FormAssociatedElement.cpp

```diff
@@ -657,6 +655,35 @@ void FormAssociatedTextControlElement::collapse_selection_to_offset(size_t posit
 
 void FormAssociatedTextControlElement::selection_was_changed()
 {
+    auto& document = form_associated_element_to_html_element().document();
+
+    // https://w3c.github.io/selection-api/#scheduling-selectionhange-event
+    // 1. If target's has scheduled selectionchange event is true, abort these steps.
+    if (!m_has_scheduled_selectionchange_event) {
+
+        // AD-HOC (https://github.com/w3c/selection-api/issues/338):
+        // Set target's has scheduled selectionchange event to true
+        m_has_scheduled_selectionchange_event = true;
+
+        // 2. Queue a task on the user interaction task source to fire a selectionchange event on target.
+        queue_global_task(HTML::Task::Source::UserInteraction, relevant_global_object(document), JS::create_heap_function(document.heap(), [this] {
+            // https://w3c.github.io/selection-api/#firing-selectionhange-event
+            auto& element = this->form_associated_element_to_html_element();
+
+            // 1. Set target's has scheduled selectionchange event to false.
+            m_has_scheduled_selectionchange_event = false;
+
+            // 2. If target is an element, fire an event named selectionchange, which bubbles and not cancelable, at target.
+            // 3. Otherwise, if target is a document, fire an event named selectionchange, which does not bubble and not cancelable, at target.
+            DOM::EventInit event_init;
+            event_init.bubbles = false;
```

Comment:
Shouldn't this bubble?

Reply:
And if so, we're probably missing a test :)

Reply:
indeed, I fixed this and made the test check if event bubbles.